### PR TITLE
feat: add command menu for navigation

### PR DESCRIPTION
**Pull request by ieedan**
Super cool project completely blowing mine out of the water!

I was checking it out on mobile and noticed this wasn't implemented yet. 

This is a pretty rough implementation but I think it looks decent and does everything Vercel's does. Unfortunately I can't make use of [cmdk-sv](https://github.com/huntabyte/cmdk-sv) here because it only works as a modal not with responsive transformation to a drawer. May be worth opening a PR for in the future.

I had to add a `magnifying-glass` icon not sure if you are just copy pasting from Vercel but thats what I did.

I put the `Command` component in the header to save some complexity of having to setup state to call it but it could be moved if needed (would just add some complexity).

### CODE REVIEW

**Comment by ieedan** [comment]
Not sure whats up with the preview doesn't seem to have any of my changes

**Comment by shyakadavis** [comment]
Hello, @ieedan 

Thank you very much for the P.R. and the kind words! Your project is inspiring and sets a high standard.

Regarding the previews not working, I first noticed it recently, even yesterday in #15. The general assumption is that something changed in either GH Actions or Cloudflare, affecting Adrian's custom action setup for Cloudflare Pages. Please bear with this as we await a resolution.

Thanks again for your contribution. Please give me a few hours (I'm about to commute) and I'll provide a proper review.

File: package.json [hunk 1]
```diff
@@ -41,8 +41,11 @@
 	"dependencies": {
 		"bits-ui": "^0.21.12",
 		"clsx": "^2.1.1",
+		"lucide-svelte": "^0.414.0",
```
Comment (shyakadavis) [inline package.json:44]: I think we should avoid pulling in this lib. At least not when we have the alternative of using the ones we need from vercel.

File: src/lib/assets/icons/index.ts [hunk 2]
```diff
@@ -38,6 +38,7 @@ import LogoTurborepo from './logo-turborepo.svg?component';
 import LogoV0 from './logo-v0.svg?component';
 import LogoVercelCircle from './logo-vercel-circle.svg?component';
 import LogoVercel from './logo-vercel.svg?component';
+import MagnifyingGlass from './magnifying-glass.svg';
```
Comment (shyakadavis) [inline src/lib/assets/icons/index.ts:41]: We are taking advantage of `@poppanator/sveltekit-svg` to be able to import `.svg` files as Svelte components.

```suggestion
import MagnifyingGlass from './magnifying-glass.svg?component';
```

File: src/lib/components/shared/command.svelte
Comment (shyakadavis) [inline src/lib/components/shared/command.svelte:1]: I have some questions about this approach for the command. It's not a change-request per se, because I appreciate the work you put into this, but noticed some minor bugs that I feel could be simplified.

When you mentioned the limitations of `cmdk-sv`, I tried implementing a draft of the `Command` component on this branch: https://github.com/shyakadavis/geist/tree/dev

IMO, at least for desktop, cmdk-sv does the job well. Without, they are some minor bugs like not closing the modal on clicking a link, and some console errors like: `Uncaught TypeError: 'close' called on an object that does not implement interface Window.`

---

As for mobile, I am curious to hear what you think about this:

On mobile, there's no keyboard, and applying a `mouse_move` function doesn't make sense, right? For mobile, I believe all we need is a filter function inside the drawer, and a click handler.

---

To sum up my questions... what if we go with cmdk-sv, and have no extra logic on mobile save for the filter function?

File: src/lib/components/ui/dialog/dialog-content.svelte [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	export let includeX: $$Props['includeX'] = true;
+</script>
+
+<Dialog.Portal>
+	<Dialog.Overlay />
+	<DialogPrimitive.Content
+		{transition}
+		{transitionConfig}
+		class={cn(
+			'bg-background fixed left-[50%] top-[15%] z-50 grid w-full max-w-lg translate-x-[-50%] gap-4 border p-6 shadow-lg sm:rounded-lg md:w-full',
```
Comment (shyakadavis) [inline src/lib/components/ui/dialog/dialog-content.svelte:26]: I failed to stick to the `shadcn-svelte` colours like `background, muted, accent...`, and as a result, class names like `bg-background` produce no output.

I tried to help out in the https://github.com/shyakadavis/geist/tree/dev branch and added the equivalent classes if you want to copy-paste.

**Review by shyakadavis — CHANGES_REQUESTED** [review]
Hey, @ieedan 

Thanks again for taking the time. Here are some suggestions about implementation direction.

Please let me know what you think.

File: package.json (hunk 1 above)
Comment (ieedan) [reply package.json:44]: Yeah no problem we aren't even using it its just there because the normal dialog component uses it.

File: src/lib/assets/icons/index.ts (hunk 2 above)
Comment (ieedan) [reply src/lib/assets/icons/index.ts:41]: Not sure how I missed this thanks

**Comment by ieedan** [comment]
I'll be more available in a few hours and I'll take another look thanks!

File: src/lib/components/shared/command.svelte
Comment (ieedan) [reply src/lib/components/shared/command.svelte:1]: > Uncaught TypeError: 'close' called on an object that does not implement interface Window.

I hadn't noticed that I will take a look.

I guess it would make the most sense to use cmdk part of the time and a drawer the rest?

File: src/lib/components/ui/dialog/dialog-content.svelte (hunk 3 above)
Comment (ieedan) [reply src/lib/components/ui/dialog/dialog-content.svelte:26]: I noticed that I'll need to purge dead classes. I think it may be good to add a part for this in a contribution guide of sorts.

Comment (shyakadavis) [reply src/lib/components/ui/dialog/dialog-content.svelte:26]: Ah, nice suggestion. I'll definitely a contribution guide tonight. Thanks.

**Comment by ieedan** [comment]
I am going to work on this now.

Is auto closing on selection not the intended behavior? Thats normally my default and what Vercel is doing here.

**Comment by ieedan** [comment]
> > Is auto closing on selection not the intended behavior?
> 
> It is. But oddly, it's not working. 🤷

Yeah sorry I am fixing that right now.

**Comment by ieedan** [comment]
Everything should be working good now. 

I had to create a context for the close function since I separated the list so I added a helper for that.

**Comment by ieedan** [comment]
Still uses dialog implementation right now but I can change to the cmdk-sv implementation if needed.

**Comment by shyakadavis** [comment]
Hey, @ieedan 

Could you merge the changes from main and see if the previews are working? (You are working off `main` on your fork, so I can't commit directly to this P.R.)

**Comment by ieedan** [comment]
I tried that earlier it says up to date

**Comment by ieedan** [comment]
> (You are working off `main` on your fork, so I can't commit directly to this P.R.)

Oops next time I'll make a branch

**Comment by ieedan** [comment]
Looks like its working TY Adrian

**Comment by ieedan** [comment]
Not sure if this is a safari thing but there are double arrows on the details component on mobile image

**Comment by ieedan** [comment]
That fix works Claude is awesome man

**Comment by shyakadavis** [comment]
Hello, @ieedan 

> That fix works Claude is awesome man

It's also my go-to tool instead of the others 😅 

I really wish you could create a new branch off of this one and open a separate P.R, because I figured out a way to simplify things using only `cmdk-sv`, and I didn't want to discard your input.

Please have a look at the https://github.com/shyakadavis/geist/tree/dev branch, because I feel like this is the best approach forward. Correct me if I'm wrong.

**Comment by ieedan** [comment]
Yeah that makes more sense. I hadn't tried adding it like that. Let me go ahead and do it that way and open a separate PR unless you think it makes more sense to just use what you already have.

**Comment by shyakadavis** [comment]
Let's go with your initial draft. If need be, I'll commit with improvements. What do you think?

**Comment by ieedan** [comment]
I don't want to merge something and make you do more work if you give me a few minutes I can refactor with `cmdk-sv`

**Comment by ieedan** [comment]
Sorry this took MUCH longer than I thought it would. Styling was a little difficult then I had a meeting and floor issue I had to address.

**Comment by shyakadavis** [comment]
Hey, @ieedan 

Thanks again for taking the time. I am happy with the current implementation, and all that's left to do is some clean-up.

Namely:

- cmd + k not triggering the command menu (tested on Firefox)
- ghost class names that render no output
- snake_casing 😅 
- double focus traps when trying to reach `esc` button
- using components like `Button` instead of raw ones
- unused files
- ...

I just tested to see if I could commit to this PR directly, as opposed to merging and opening a follow-up P.R, and surprisingly, it works, save for having to manually trigger workflows to run.

I don't want to take up more of your time, nor hijack your P.R, but with your permission, I would like to take it across the finish line.

What do you think?

**Comment by ieedan** [comment]
If that's what you'd prefer. I'm sorry for wasting so much of your time with this 😅. 

I noticed the snake case thing I am just so used to camel for JS (even though snake case is better). Maybe worth adding eslint to check that I think there are ways you can enforce it.

I still have time to work on this so I can try and tackle things in the next hour? Still don't want to waste any more of your time so if you'd rather just do it I understand completely.

**Comment by shyakadavis** [comment]
Please don’t worry about wasting my time—every contribution is greatly appreciated.

Regarding casing, I'll definitely set up ESLint tomorrow.

If you still have time and energy to work on this, please go ahead!

**Comment by ieedan** [comment]
## Issue Tracker
- [x] CTRL + K Firefox
- [ ] Ghost class names
- [x] Double focus traps
- [x] No more raw dog buttons
- [x] Unused files

**Comment by ieedan** [comment]
Okay I think that is everything.

FF is so dumb sometimes I swear why do I have to stopProp and preventDef

**Comment by ieedan** [comment]
It'll probably be worth squashing this one for a cleaner history.

Sorry about that 😅

**Comment by shyakadavis** [comment]
These tweaks are in the spirit of keeping the `ui` components intact in case we wish to re-use them elsewhere. As such, specific overrides ought to happen at the consumer's end, rather than the component definition itself.

As for focus and cmd+k, here is a video of what I was seeing before.


https://github.com/user-attachments/assets/6c935006-35a9-4279-a3b5-80433d1417f5

Should be resolved on my end, would appreciate confirmation that I didn't mess up anything on your end.

**Review by shyakadavis — APPROVED** [review]
Thank you very much @ieedan
